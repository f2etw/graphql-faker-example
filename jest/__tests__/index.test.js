import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Index from '../../web/pages';

configure({ adapter: new Adapter() });

const client = {
  query: jest.fn(() => Promise.resolve()),
  refetch: jest.fn(() => Promise.resolve()),
  watchQuery: (...request) => ({
    currentResult: () => client.query(...request),
    refetch: () => client.refetch(...request),
    subscribe: jest.fn(),
  }),
  mutate: jest.fn(() => Promise.resolve()),
  fetch: jest.fn(() => Promise.resolve()),
};

describe('index', () => {
  it('successfully render', () => {
    client.query.mockReturnValueOnce({ me: { id: '123', name: 'yutin' } });
    const component = shallow(<Index />, { context: client });
    expect(component).toMatchSnapshot();
  });
});
