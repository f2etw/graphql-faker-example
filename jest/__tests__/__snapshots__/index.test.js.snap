// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index successfully render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithData(Apollo(Component))
    cache={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Apollo(Component)
        cache={Object {}}
      />,
      "client": ApolloClient {
        "cache": InMemoryCache {
          "addTypename": true,
          "config": Object {
            "addTypename": true,
            "dataIdFromObject": [Function],
            "fragmentMatcher": IntrospectionFragmentMatcher {
              "isReady": false,
              "match": [Function],
            },
            "storeFactory": [Function],
          },
          "data": ObjectCache {
            "data": Object {},
          },
          "optimistic": Array [],
          "silenceBroadcast": false,
          "watches": Array [],
        },
        "defaultOptions": Object {},
        "disableNetworkFetches": true,
        "link": ApolloLink {
          "request": [Function],
        },
        "mutate": [Function],
        "query": [Function],
        "queryDeduplication": true,
        "reFetchObservableQueries": [Function],
        "resetStore": [Function],
        "ssrMode": true,
        "store": DataStore {
          "cache": InMemoryCache {
            "addTypename": true,
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "fragmentMatcher": IntrospectionFragmentMatcher {
                "isReady": false,
                "match": [Function],
              },
              "storeFactory": [Function],
            },
            "data": ObjectCache {
              "data": Object {},
            },
            "optimistic": Array [],
            "silenceBroadcast": false,
            "watches": Array [],
          },
        },
        "version": "2.0.4",
        "watchQuery": [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "cache": Object {},
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Apollo(Component)
          cache={Object {}}
        />,
        "client": ApolloClient {
          "cache": InMemoryCache {
            "addTypename": true,
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "fragmentMatcher": IntrospectionFragmentMatcher {
                "isReady": false,
                "match": [Function],
              },
              "storeFactory": [Function],
            },
            "data": ObjectCache {
              "data": Object {},
            },
            "optimistic": Array [],
            "silenceBroadcast": false,
            "watches": Array [],
          },
          "defaultOptions": Object {},
          "disableNetworkFetches": true,
          "link": ApolloLink {
            "request": [Function],
          },
          "mutate": [Function],
          "query": [Function],
          "queryDeduplication": true,
          "reFetchObservableQueries": [Function],
          "resetStore": [Function],
          "ssrMode": true,
          "store": DataStore {
            "cache": InMemoryCache {
              "addTypename": true,
              "config": Object {
                "addTypename": true,
                "dataIdFromObject": [Function],
                "fragmentMatcher": IntrospectionFragmentMatcher {
                  "isReady": false,
                  "match": [Function],
                },
                "storeFactory": [Function],
              },
              "data": ObjectCache {
                "data": Object {},
              },
              "optimistic": Array [],
              "silenceBroadcast": false,
              "watches": Array [],
            },
          },
          "version": "2.0.4",
          "watchQuery": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "cache": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "fetch": [MockFunction],
      "mutate": [MockFunction],
      "query": [MockFunction],
      "refetch": [MockFunction],
      "watchQuery": [Function],
    },
  },
}
`;
